lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php

```diff
@@ -527,6 +530,9 @@ protected function doWalkConstraintComposite(AbstractNode $node, $method)
 
             $lPhpcrConstraint = $phpcrComposite;
         }
+        if (!$phpcrComposite) {
+            throw new \InvalidArgumentException('Composite must have at least one constraint');
+        }
```

Comment:
this is an actual code change!
my IDE warned that $phpcrComposite might be undefined. i think this is the correct behaviour, as we would have gotten a fatal error before when there are no constraints.